graveler/ref/branch_iterator.go

```diff
@@ -8,22 +8,17 @@ import (
 )
 
 type BranchIterator struct {
-	db  db.Database
-	ctx context.Context
-
+	db           db.Database
+	ctx          context.Context
 	repositoryID graveler.RepositoryID
 	value        *graveler.BranchRecord
 	buf          []*graveler.BranchRecord
-
-	// blockValue is true when the iterator is created, or SeekGE() is called.
-	// A single Next() call turns it to false. When it's true, Value() returns nil.
-	blockValue bool
-
-	offset      string
-	fetchSize   int
-	shouldFetch bool
-
-	err error
+	offset       string
+	fetchSize    int
+	err          error
+	fetchCalled  bool
+	fetchEnded   bool
+	closed       bool
```

Comment:
Thanks, good names!  Now I see and understand: _it's a state machine_!  Not in this PR, but I would like to try refactor it to have one explicit state.  Think it's worth a shot?